### DISCUSSION TITLE
Vue 3: Composition API

### DISCUSSION THREAD

**Discussion by ilyaDegtyarenko** [advanced-cropper/vue-advanced-cropper#197] [General]
How to use with composition api? (vue 3)

P.S. Nuxt 3

**Comment by Norserium** [comment]
@ilyaDegtyarenko, I don't see any thing that would be specific for this cropper if you will use Composition API. Try, for example, [this sandbox](https://codesandbox.io/s/vue-advanced-cropper-composition-api-5z0ww0?file=/src/App.vue).

**Reply by ilyaDegtyarenko** [reply]
Thank you, you are right

**Reply by Norserium** [reply]
@ilyaDegtyarenko, you are welcome!

**Comment by ghost** [comment]
@Norserium 
I would like to express my gratitude for the example code that you provided.However, I am currently facing an issue with manipulating, rotating, and flipping images using the Composition API.
I tried using the following code snippet:
```
const cropper = ref(null)
const zoom = () => {
    cropper.zoom(2)
}
const flip = (x, y) => {
    cropper.flip(x, y)
}
const rotate = (angle) => {
    cropper.rotate(angle)
}
```
Unfortunately, it did not work as expected. Could you please provide some guidance on how to properly manipulate, rotate, and flip images using the Composition API in Vue 3? Any help or suggestions would be greatly appreciated.

**Reply by Norserium** [reply]
There is [the documentation about `ref` in Composition API](https://vuejs.org/api/reactivity-core.html#ref):
> Takes an inner value and returns a reactive and mutable ref object, which has a single property .value that points to the inner value.

So, to get the actual ref to the cropper, you should write, for example:
```
cropper.value.rotate(angle)
```

**Reply by ghost** [reply]
Sorry, I overlooked some common mistakes. However, even after adding the .value, I still cannot use zoom/rotate/flip. I mean, the relevant buttons are not appearing in the Cropper component. Are additional parameters needed to invoke these methods? Looking forward to your reply, thank you very much.
```
const cropper= ref(null)
const zoom = () => {
    cropper.value.zoom(2)
}
```

**Reply by Norserium** [reply]
@orzcyt, did you pass `cropper` below to the component and use it in your template? Log `cropper.value` before you use it.

**Reply by ghost** [reply]
I followed the example from the [Docs](https://advanced-cropper.github.io/vue-advanced-cropper/guides/advanced-recipes.html#manipulate-image):

```
<Cropper
    ref="cropper"
    :src="image.src"
    :stencil-size="{
        width: 255,
        height: 255,
    }"
    :canvas="{
        height: 255,
        width: 255,
    }"
    @change="resule"
/>
```
How should I use it in the template? I want to achieve the zoom buttons in the example, is there any related example for that? Thank you again.

**Reply by Norserium** [reply]
@orzcyt, I've created [the example ](https://codesandbox.io/s/vue-advanced-cropper-zoom-mflhd7?file=/src/App.vue) for you.

**Reply by ghost** [reply]
Thank you for your reply, I understand that these buttons need to be created manually outside. In this [Docs example](https://advanced-cropper.github.io/vue-advanced-cropper/guides/advanced-recipes.html#manipulate-image), it looks like these buttons are built-in to the Cropper component, and the example does not give a way to implement them, which may cause some confusion. It would be better if you could write more details. Thank you again for your contribution.

**Reply by Norserium** [reply]
@orzcyt, thank you for your feedback. I always appreciate it, because I can't see my examples by user eyes. Is your issue solved now?

**Reply by ghost** [reply]
I have solved the problem. It would be nice if cropper could embed the buttons into the cropper component like in the example, using slot or something similar.
